src/transformers/testing_utils.py

```diff
@@ -1570,6 +1572,38 @@ def LoggingLevel(level):
         transformers_logging.set_verbosity(orig_level)
 
 
+class TemporaryHubRepo:
+    """Create a temporary Hub repository and return its `RepoUrl` object. This is similar to
+    `tempfile.TemporaryDirectory` and can be used as a context manager.  For example:
+
+        with TemporaryHubRepo() as temp_repo:
+            ...
+
+    Upon exiting the context, the repository and everything contained in it are removed.
+
+    Example:
+
+    ```python
+    with TemporaryHubRepo() as temp_repo:
+        model.push_to_hub(tmp_repo.repo_id, token=self._token)
```

Comment:
```suggestion
    with TemporaryHubRepo(token=self._token) as temp_repo:
        model.push_to_hub(tmp_repo.repo_id, token=self._token)
```

(token should be passed to both calls IIUC)